sdks/go/pkg/beam/windowing.go

```diff
@@ -21,21 +21,48 @@ import (
 	"github.com/apache/beam/sdks/go/pkg/beam/internal/errors"
 )
 
+type WindowIntoOption interface {
+	windowIntoOption()
+}
+
+type WindowTrigger struct {
+	Name window.Trigger
+}
+
+func (t WindowTrigger) windowIntoOption() {}
+
+type AccumulationMode struct {
+	Mode window.AccumulationMode
+}
+
+func (m AccumulationMode) windowIntoOption() {}
+
 // WindowInto applies the windowing strategy to each element.
-func WindowInto(s Scope, ws *window.Fn, col PCollection) PCollection {
-	return Must(TryWindowInto(s, ws, col))
+func WindowInto(s Scope, ws *window.Fn, col PCollection, opts ...WindowIntoOption) PCollection {
+	return Must(TryWindowInto(s, ws, col, opts...))
 }
 
 // TryWindowInto attempts to insert a WindowInto transform.
-func TryWindowInto(s Scope, ws *window.Fn, col PCollection) (PCollection, error) {
+func TryWindowInto(s Scope, wfn *window.Fn, col PCollection, opts ...WindowIntoOption) (PCollection, error) {
 	if !s.IsValid() {
 		return PCollection{}, errors.New("invalid scope")
 	}
 	if !col.IsValid() {
 		return PCollection{}, errors.New("invalid input pcollection")
 	}
+	ws := window.WindowingStrategy{Fn: wfn, Trigger: window.Trigger{}}
+	for _, opt := range opts {
+		switch opt := opt.(type) {
+		case WindowTrigger:
+			ws.Trigger = opt.Name
+		case AccumulationMode:
+			ws.AccumulationMode = opt.Mode
+		default:
+			panic("Invalid option for Windowing Strategy")
```

Comment:
Since WindowIntoOptions are implementers of interfaces, we can improve this by actually communicating what's going wrong. That is something like `panic(fmt.Sprintf("Unknown WindowingInto option type: %T: %v", opt, opt))` which communicates the type, and if it's printable somehow so it's easier to find what needs working on.

Since an error like this is likely an SDK dev error, a panic is fine, as there's nothing a user can do about it programmatically.